keras/src/models/functional_test.py

```diff
@@ -705,12 +704,17 @@ def test_list_input_with_dict_build(self):
         x2 = Input((10,), name="IS")
         y = layers.subtract([x1, x2])
         model = Model(inputs={"IT": x1, "IS": x2}, outputs=y)
-        x1 = ops.ones((1, 10))
-        x2 = ops.zeros((1, 10))
+        x1 = np.ones((1, 10))
+        x2 = np.zeros((1, 10))
         # Works
         _ = model({"IT": x1, "IS": x2})
-        with self.assertRaisesRegex(
-            ValueError,
-            "The structure of `inputs` doesn't match the expected structure",
-        ):
-            model([x1, x2])
+        # Deprecated ValueError behavior: Works with support for dict-inputs.
```

Comment:
Do we really want this behavior? This might cause a lot of confusion with respect to the order of inputs. Which order is applied?

For example: https://github.com/keras-team/keras/issues/20070